extensions/amp-access/0.1/test/test-access-expr.js

```diff
@@ -194,6 +194,65 @@ describe('evaluateAccessExpr', () => {
     expect(evaluateAccessExpr('a AND c', {a: 1, b: 2})).to.be.false;
   });
 
+  it('should evaluate nested expressions', () => {
+    const resp = {
+      obj: {
+        str: 'A',
+        num: 11,
+        bool: true,
+      }
+    };
+
+    expect(evaluateAccessExpr('obj.bool = true', resp)).to.be.true;
+    expect(evaluateAccessExpr('obj.num = 11', resp)).to.be.true;
+    expect(evaluateAccessExpr('obj.str = "A"', resp)).to.be.true;
+
+    expect(evaluateAccessExpr('obj.other = NULL', resp)).to.be.true;
+    expect(evaluateAccessExpr('obj.str = NULL', resp)).to.be.false;
+
+    expect(evaluateAccessExpr('obj.bool', resp)).to.be.true;
+    expect(evaluateAccessExpr('obj.str', resp)).to.be.true;
+    expect(evaluateAccessExpr('obj.num', resp)).to.be.true;
+    expect(evaluateAccessExpr('obj.other', resp)).to.be.false;
+
+    expect(evaluateAccessExpr('NOT obj.bool', resp)).to.be.false;
+    expect(evaluateAccessExpr('NOT obj.str', resp)).to.be.false;
+    expect(evaluateAccessExpr('NOT obj.num', resp)).to.be.false;
+    expect(evaluateAccessExpr('NOT obj.other', resp)).to.be.true;
+  });
+
+  it('should shortcircuit nested expressions with missing parent', () => {
+    const resp = {
+      obj: {
+        str: 'A',
+        child: {
+          str: 'B'
+        }
+      },
+    };
+
+    expect(evaluateAccessExpr('obj.str = "A"', resp)).to.be.true;
+    expect(evaluateAccessExpr('obj.child.str = "B"', resp)).to.be.true;
+    expect(evaluateAccessExpr('obj.child.other = NULL', resp)).to.be.true;
+    expect(evaluateAccessExpr('obj.child2 = NULL', resp)).to.be.true;
+    expect(evaluateAccessExpr('obj.child2.other = NULL', resp)).to.be.true;
```

Comment:
lets add a test for `NULL` being on left hand side used as a NAME. (ex. obj.NULL.other to throw (?))